src/sentry/api/invite_helper.py

```diff
@@ -1,33 +1,45 @@
 from __future__ import annotations
 
 from logging import Logger
-from typing import Any, Dict, Tuple
+from typing import Dict, Optional, Tuple
 
 from django.utils.crypto import constant_time_compare
 from rest_framework.request import Request
 
 from sentry import audit_log, features
-from sentry.models import AuthIdentity, AuthProvider, OrganizationMember, User, UserEmail
-from sentry.signals import member_joined
+from sentry.models import AuthIdentity, AuthProvider, User, UserEmail
+from sentry.services.hybrid_cloud.organization import (
+    RpcOrganizationMember,
+    RpcUserInviteContext,
+    organization_service,
+)
 from sentry.utils import metrics
 from sentry.utils.audit import create_audit_entry
 
 
-def add_invite_details_to_session(request: Request, member_id: int, token: str) -> None:
+def add_invite_details_to_session(
+    request: Request, member_id: int, token: str, organization_id: int
+) -> None:
     """Add member ID and token to the request session"""
     request.session["invite_token"] = token
     request.session["invite_member_id"] = member_id
+    request.session["invite_organization_id"] = organization_id
 
 
 def remove_invite_details_from_session(request: Request) -> None:
     """Deletes invite details from the request session"""
     request.session.pop("invite_member_id", None)
     request.session.pop("invite_token", None)
+    request.session.pop("invite_organization_id", None)
 
 
-def get_invite_details(request: Request) -> Tuple[str, int]:
+def get_invite_details(request: Request) -> Tuple[Optional[str], Optional[int], Optional[int]]:
```

Comment:
Could we introduce a small dataclass to wrap these three values?

Alternatively, just make the function private (i.e., rename to `_get_invite_details`). I'm more okay with weird tuples if they don't leave the module.

Reply:
Sure, that's simple enough.